src/components/ListItem.jsx

```diff
@@ -1,5 +1,57 @@
 import './ListItem.css';
+import { updateItem } from '../api/firebase';
 
-export function ListItem({ name }) {
-	return <li className="ListItem">{name}</li>;
+const currentTime = new Date();
+const numOfSecondsinMin = 60;
+const millisecondsToSeconds = 1000;
+
+export function ListItem({ name, item, listToken }) {
+	const { isChecked, id, totalPurchases, dateLastPurchased } = item;
+
+	if (isChecked) {
+		//check if it's been 24 hours since item was last marked as purchased
+		const timeDiffMinutes =
```

Comment:
I'm so curious to hear how you arrived at this method for the 24 hour time period! Were you able to verify that it's working?

Reply:
Google! lol and we were able to test it. We checked a few items as purchased and changed their purchase date in the FireStore to ensure that it was working properly.

Reply:
Clever!